### PR TITLE
chore: replace `tsx` with swc

### PR DESCRIPTION
**Pull request by SukkaW**
The PR replaces `tsx` with `@swc-node/register`.

`bunchee` already prefers swc, it doesn't make sense to use esbuild as the build script loader.  Let's switch it to swc.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "clean": "rm -rf ./dist",
     "typecheck": "tsc --noEmit && tsc -p test/tsconfig.json --noEmit",
     "prepublishOnly": "pnpm clean && pnpm build && chmod +x ./dist/bin/cli.js && pnpm test",
-    "build": "tsx ./src/bin/index.ts --runtime node",
+    "build": "SWC_NODE_IGNORE_DYNAMIC=true node -r @swc-node/register ./src/bin/index.ts --runtime node",
```
Comment (SukkaW) [inline package.json:15]: `SWC_NODE_IGNORE_DYNAMIC` is introduced in https://github.com/swc-project/swc-node/pull/745.

**Comment by huozhi** [comment]
We were also using tsx in tests, could you also replace them together? thanks!

**Comment by SukkaW** [comment]
> Some tests are failing

I am still experimenting with things here and there. I am tracking CI at here: https://github.com/SukkaW/bunchee/pulls

**Comment by huozhi** [comment]
Looks good, is `SWC_NODE_IGNORE_DYNAMIC` still required?

**Comment by SukkaW** [comment]
> Looks good, is `SWC_NODE_IGNORE_DYNAMIC` still required?

Yes.

By default, SWC transforms all dynamic imports into `Promise.resolve(require())`, which disrupts the importing of pure ESM modules in CommonJS. To maintain the original dynamic import syntax, the `SWC_NODE_IGNORE_DYNAMIC` environment variable is necessary.

**Comment by huozhi** [comment]
I mean we don't have any case like that in bunchee so probably don't need it. I'll land this since everything looks good! Thanks!

It sounds like a good feature to add into swc compilation. e.g. if the input is a cjs file and using dynamic import, SWC should preserve it. Are we having this in swc plugin? @SukkaW

**Pull request merged**